pynxtools_stm/stm_file_parser.py

```diff
@@ -317,18 +309,36 @@ def get_dimension_info(self, config_dict, data_dict):
             length on (x, y)-dimenstion and one last unknown values.
         """
         scanfield: str = ""
+        scan_range: str = ""
+        scan_offset: str = ""
+        scintific_num_pattern = r"[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?"
         for key, val in config_dict.items():
             if (
                 "/ENTRY[entry]/INSTRUMENT[instrument]/ENVIRONMENT[environment]/"
                 "scan_control/positioner/scanfield"
             ) == key:
                 if val in data_dict:
                     scanfield = data_dict[val]
-                else:
-                    raise ValueError(
-                        "Scanfield data missing: /ENTRY[entry]/INSTRUMENT[instrument]"
-                        "/ENVIRONMENT[environment]/scan_control/positioner/scanfield"
-                    )
+            elif (
+                "/ENTRY[entry]/INSTRUMENT[instrument]/ENVIRONMENT[environment]"
+                "/scan_control/scan_range"
+            ) == key:
+                scan_range = data_dict.get(val, None)
+                if scan_range:
+                    scan_range = re.findall(scintific_num_pattern, scan_range)
+                    template[key] = to_intended_t(scan_range)
+            elif (
+                "/ENTRY[entry]/INSTRUMENT[instrument]/ENVIRONMENT[environment]"
+                "/scan_control/scan_offset"
+            ) == key:
+                scan_offset = data_dict.get(val, None)
+                if scan_offset:
+                    scan_offset = re.findall(scintific_num_pattern, scan_offset)
+                    template[key] = to_intended_t(scan_offset)
+        if (not scan_offset or not scan_range) and not scanfield:
+            raise KeyError(
+                "Scanfield, scan_range and scan_offset are not found in raw data file."
```

Comment:
```suggestion
                "Scanfield, scan_range, and scan_offset are not found in raw data file."
```
Based on the if clause, the conditions is a bit different. It is either scanfield and scan_offset OR scanfield and scan_range are not found. But probably okay like this.

Reply:
Noted!